servers/cromwell/jobs/controllers/jobs_controller.py

```diff
@@ -206,6 +206,9 @@ def query_jobs(body, **kwargs):
         offset = offset + page_size
         page = page_from_offset(offset, page_size)
 
+    if query_page_size > len(results):
```

Comment:
This still mishandles one edge case which is where the last page is exactly the length of the page size. You don't want to return a next_page_token in that case.

I'm not familiar with this particular code, but typically you want to try and collect query_page_size+1 elements, then truncate to page_size on return, populating next_page_token only if you see the n+1th element.

Reply:
The above loop breaks at query_page_size, so now there's a new bug which is that you might not return a page token when there are still more jobs. See the second part of my above comment.